src/torchcodec/decoders/_video_decoder.py

```diff
@@ -379,3 +394,40 @@ def _get_and_validate_stream_metadata(
         end_stream_seconds,
         num_frames,
     )
+
+
+def read_custom_frame_mappings(
+    custom_frame_mappings: Union[bytes, bytearray, str]
+) -> tuple[Tensor, Tensor, Tensor]:
+    try:
+        if hasattr(custom_frame_mappings, "read"):
+            input_data = json.load(custom_frame_mappings)
+        else:
+            input_data = json.loads(custom_frame_mappings)
+    except json.JSONDecodeError:
+        raise ValueError(
+            "Invalid custom frame mappings. "
+            "It should be a valid JSON string or a JSON file object."
```

Comment:
In our type annotation, we only say we accept string-like objects, not file-like objects. In other parts of the code, I've indicated a file-like object with `io.RawIOBase` and `io.BufferedReader`: https://github.com/pytorch/torchcodec/blob/c3eea9f9f42d3d51f9c53ba19be6c11cc88c21dd/src/torchcodec/decoders/_video_decoder.py#L74-L77

For our doc string, however, we just say "file-like object" since those actual types are probably meaningless to most users: https://github.com/pytorch/torchcodec/blob/c3eea9f9f42d3d51f9c53ba19be6c11cc88c21dd/src/torchcodec/decoders/_video_decoder.py#L26

Which introduces a good point: since this a top-level API that we expect users to call, we need a docstring. :)